### ISSUE TITLE
[analyzer] core.UndefinedBinaryOperatorResult false positive on C arrays cast

### ISSUE THREAD

**Issue by hedrok** [llvm/llvm-project#168597]
```c
#include <stdio.h>

int main()
{
    unsigned int ints[2] = {0xaabbccdd, 0xff};
    unsigned char* bytes = (unsigned char*)ints;
    printf("0x%x\n", bytes[1] | 1);
}
```

Calling
```bash
scan-build clang false-positive.c
```

Gives

```
scan-build: Using '/usr/lib/llvm-22/bin/clang' for static analysis
false-positive.c:7:31: warning: The left operand of '|' is a garbage value [core.UndefinedBinaryOperatorResult]
    7 |     printf("0x%x\n", bytes[1] | 1);
      |                      ~~~~~~~~ ^
1 warning generated.
scan-build: Analysis run complete.
scan-build: 1 bug found.
scan-build: Run 'scan-view /tmp/scan-build-2025-11-18-191927-81-1' to examine bug reports.
```

Checked for versions:

```
Debian clang version 22.0.0 (++20251015112211+4ad625b15bc1-1~exp1~20251015112229.1767)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/lib/llvm-22/bin
```

And

```
Debian clang version 19.1.7 (++20250114103228+cd708029e0b2-1~exp1~20250114103334.78)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/lib/llvm-19/bin
```

**Comment by NagyDonat** [comment]
🤔 Fair point, the character value is not undefined (=garbage), but only implementation-defined.

I agree that this shouldn't be reported by the `core.UndefinedBinaryOperatorResult` checker and analogous checkers, but if ideally there should be a checker (probably a new one, e.g. `optin.EndiannesDependentValues`) which reports code like this.

My first instinct is that in this "read a byte from the middle of a number" situation (where the analyzer currently returns Undefined) the result should be either a `SymbolConjured` or a new kind of symbol that represents "$n$th byte of _this_ number". (We cannot return a concrete number because it depends on endianness.)
